docs(chart): Add documentation for range series types (#1536)
telerik/blazor-docs: components/chart/types/rangearea.md

---
title: Range Area
page_title: Chart - Range Area

> description: Overview of the Range Area Chart for Blazor.

Reviewer: Try to write descriptions that are around 150 symbols and describe the content of the article.

> description: Overview of the Range Area Chart for Blazor.

Author: ```suggestion
description: Overview of the Range Area Chart for Blazor with a description of the common use cases and the different ways to data bind the chart. The article lists configuration options and provides Range Area Chart examples.
```

slug: components/chart/types/rangearea
tags: telerik,blazor,chart,rangearea
published: True
position: 0
---

# Range Area Chart

> The <a href="https://www.telerik.com/blazor-ui/range-area-chart" target="_blank">Blazor Range Area Chart</a> shows the data as a colored area between two continuous lines that pass through points defined by their items' "low" and "high" values. The graph between the border lines has a different customizable color for each series. The Range Area Chart is similar to the [Area Chart]({%slug components/chart/types/area%}). The latter can be regarded as a Range Area Chart with a "low" value of zero.

Author: I used the quotes to highlight that these are non-official terms. There is a way to improve this.

> The <a href="https://www.telerik.com/blazor-ui/range-area-chart" target="_blank">Blazor Range Area Chart</a> shows the data as a colored area between two continuous lines that pass through points defined by their items' "low" and "high" values. The graph between the border lines has a different customizable color for each series. The Range Area Chart is similar to the [Area Chart]({%slug components/chart/types/area%}), which can be regarded as a Range Area Chart with a constant "low" value of zero.

Author: ```suggestion
The <a href="https://www.telerik.com/blazor-ui/range-area-chart" target="_blank">Blazor Range Area Chart</a> shows the data as a colored area between two continuous lines that pass through points defined by pairs of `from` and `to` values. The graph between the border lines has a different customizable color for each series. The Range Area Chart is similar to the [Area Chart]({%slug components/chart/types/area%}), which can be regarded as a Range Area Chart with a constant `from` value of zero.
```

You can use the Range Area Chart to emphasize the difference between pairs of continuous value sequences.

By default, the series backgrounds are semi-transparent, which lets the user clearly see where different sets of data overlap.

@[template](/_contentTemplates/chart/link-to-basics.md#understand-basics-and-databinding-first)

## Creating Blazor Range Area Chart

1. Add a `ChartSeries` to the `ChartSeriesItems` collection.

> 2. Set its `Type` property to `ChartSeriesType.RangeArea`.

Author: ```suggestion
2. Set the series `Type` parameter to `ChartSeriesType.RangeArea`.
```

3. Provide a data collection to its `Data` property. You can use a [collection of arrays](#bind-range-area-series-to-collection-of-arrays) or a [collection of custom objects](#bind-range-area-series-to-custom-objects).
4. If the Range Area data is a collection of arrays, provide data for the `Categories` parameter of the `ChartCategoryAxis`.

### Bind Range Area Series to Collection of Arrays

> In this case, set the `ChartSeries` `Data` parameter to a `List` of arrays or a jagged array (an array of arrays). The inner arrays should have two members - one for the "low" ("from") value, and one for the "high" ("to") value.

Author: ```suggestion
In this case, set the `ChartSeries` `Data` parameter to a `List` of arrays or a jagged array (an array of arrays). The inner arrays should have two members - one for the lower `from` value, and one for the higher `to` value.
```

    <ChartLegend Position="ChartLegendPosition.Right"></ChartLegend>
</TelerikChart>

@code {
    private List<AreaDataPoint> Tube1Data { get; set; } = new List<AreaDataPoint>();

    private List<AreaDataPoint> Tube2Data { get; set; } = new List<AreaDataPoint>();

    protected override void OnInitialized()
    {
        var rnd = new Random();
        var dataPointCount = 10;

        for (int i = 1; i <= dataPointCount; i++)
        {
            Tube1Data.Add(new AreaDataPoint()
            {
                Hour = i,
                LowValue = rnd.Next(5 * i, 10 * i),
                HighValue = rnd.Next(15 * i, 20 * i + 5)
            });

            Tube2Data.Add(new AreaDataPoint()
            {
                Hour = i,
                LowValue = rnd.Next(5 * (dataPointCount + 1 - i), 10 * (dataPointCount + 1 - i)),
                HighValue = rnd.Next(15 * (dataPointCount + 1 - i), 20 * (dataPointCount + 1 - i))
            });
        }
    }

    public class AreaDataPoint
    {
        public int Hour { get; set; }
        public int LowValue { get; set; }
        public int HighValue { get; set; }
    }
}
````

> ## Range Area Chart Specific Appearance Settings

Reviewer: Use text between a heading and its subheading.

### Color

The color of a series is controlled through the `Color` property that can take any valid CSS color (for example, `#abcdef`, `#f00`, or `blue`). The color control the fill color of the area.

You can control the color of the line itself separately by using the `Color` property of the nested `TelerikChartSeriesLine` tag.

@[template](/_contentTemplates/chart/link-to-basics.md#opacity-area-bubble)

### Missing Values

If *both* values in a range pair are missing or `null`, you can have the Chart work around this by setting the `MissingValues` property of the series to the desired behavior. Use a member of the `Telerik.Blazor.ChartSeriesMissingValues` enum:

* `Zero` - the default behavior. The two lines and the area between them will go to the 0 value mark.
* `Interpolate` - the lines and area will go through the interpolated values of the missing data points and connect to the next data points with a value.

> * `Gap` - there will be empty space in the range area until the next pair of values.

Reviewer: ```suggestion
* `Gap`—The range area will contain empty space until the next pair of values.
```

### Line Style

> You can render the lines between the points with different styles. The supported styles can be set via the `Style` property of the child `ChartSeriesLine` tag - it takes a member of `Telerik.Blazor.ChartSeriesLineStyle` enum:

Reviewer: ```suggestion
You can render the lines between the points with different styles. The supported styles can be set via the `Style` property of the child `ChartSeriesLine` tag—it takes a member of the `Telerik.Blazor.ChartSeriesLineStyle` enum:
```

> * `Normal`—This is the default style. It produces a straight line between data points.

Reviewer: ```suggestion
* `Normal` (default)—This style produces a straight line between data points.
```
